src/android/java/io/jxcore/node/BtToRequestSocket.java

```diff
@@ -165,82 +150,89 @@ public void StreamCopyError(StreamCopyingThread who, String error) {
         }
     }
 
-    public int GetLocalHostPort() {
+    private  int GetLocalHostPort() {
         return mHTTPPort;
     }
 
-    public String GetLocalHostAddress() {
-        String ret = "";
-        if(localHostSocket != null && localHostSocket.getInetAddress() != null){
-            ret = localHostSocket.getInetAddress().toString();
-        }
-        return ret;
+    private String GetLocalHostAddressAsString() {
+        return localHostSocket == null || localHostSocket.getInetAddress() == null ? null : localHostSocket.getInetAddress().toString();
     }
 
     public void Stop() {
         mStopped = true;
```

Comment:
I did mark it as volatile. Basically Stop can be called multiple times, and there is no problem calling it before start(). The run() is called by the API. And supposedly  the API is designed in a way that it will only be called once.